Add: hr_timesheet_activity_begin_end

guewen opened the pull request:
This module adds 2 fields on timesheet activities: begin and end hours.

Commit:
Bootstrap the module hr_timesheet_activity_start_end

Commit:
Add the start and stop fields

Commit:
Prevent lines to overlap

Commit:
Rename hour_start/hour_stop to time_start/time_stop

Commit:
Sort the timesheet lines by date, then hours

Commit:
Rename module to hr_timesheet_activity_begin_end

Commit:
Exercise the module with tests

Commit:
Add translations

anthony-muschang commented:
:+1:  Code review and test

gurneyalex commented on hr_timesheet_activity_begin_end/models/account_analytic_line.py:
```diff
@@ -0,0 +1,135 @@
+            ('time_stop', '>', self.time_stop),
+            '&',
+            ('time_start', '>', self.time_start),
+            ('time_stop', '<', self.time_stop),
```
Theorem: given 2 segments S1 = [A1, B1] and S2 = [A2, B2], S1 and S2 overlap if and only if A1 < B2 && A2 < B1

Demonstration

case 1:

```
A1             B1
|-------------|
       |--------------|
       A2              B2
```

Case 2:

```
A1                             B1
|---------------------------|
       |--------------|
       A2              B2
```

And the expression is symetric which is nice because if S1 overlaps S2 then obviously S2 overlaps S1. 

Could you please simplify the domain with this in mind? I believe the resulting expression will be much simpler.

gurneyalex commented:
:+1: once remark above is implemented

Commit:
Simplified the overlap search domain

According to @afayolle's suggestion:

Theorem: given 2 segments S1 = [A1, B1] and S2 = [A2, B2], S1 and S2 overlap if
and only if A1 < B2 && A2 < B1

Demonstration

```
case 1:

    A1             B1
    |-------------|
           |--------------|
           A2              B2

Case 2:

    A1                             B1
    |---------------------------|
           |--------------|
           A2              B2
```

And the expression is symmetric which is nice because if S1 overlaps S2 then
obviously S2 overlaps S1.

guewen commented:
@gurneyalex Done! Thanks for the suggestion, this is far simpler indeed.

gurneyalex commented:
red travis build caused by #33

mdietrichc2c commented:
Travis is green now, LGTM :+1:
